File: 506. Relative Ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        answer = []
        for i in range(len(score)):
            answer.append(i)
        answer.sort(key=lambda x: score[x], reverse=True)
        result = [""] * len(answer)
        for i in range(len(answer)):
            if i == 0:
                result[answer[0]] = "Gold Medal"
            elif i == 1:
                result[answer[1]] = "Silver Medal"
            elif i == 2:
                result[answer[2]] = "Bronze Medal"
            else:
                result[answer[i]] = str(i+1)
        return result
        

File: 506. Relative Ranks.go
func findRelativeRanks(score []int) []string {
	answer := make([]int, len(score))
	for i := range score {
		answer[i] = i
	}
	sort.Slice(answer, func(i, j int) bool {
		return score[answer[i]] > score[answer[j]]
	})
	result := make([]string, len(score))
	for i := range answer {
		switch i {
		case 0:
			result[answer[i]] = "Gold Medal"
		case 1:
			result[answer[i]] = "Silver Medal"
		case 2:
			result[answer[i]] = "Bronze Medal"
		default:
			result[answer[i]] = strconv.Itoa(i + 1)
		}
	}
	return result
}
